standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1418,13 +1453,17 @@ public void commitTxn(CommitTxnRequest rqst) throws NoSuchTxnException, TxnAbort
 
     boolean isReplayedReplTxn = TxnType.REPL_CREATED.equals(rqst.getTxn_type());
     boolean isHiveReplTxn = rqst.isSetReplPolicy() && TxnType.DEFAULT.equals(rqst.getTxn_type());
+    TransactionStatus status = null;
     try {
+      //start a new transaction
+      status = jdbcTemplate.getTransactionWithinRetryContext(new DefaultTransactionDefinition(TransactionDefinition.PROPAGATION_REQUIRED), POOL_TX);
```

Comment:
why do we need transactioin with retry context if `checkRetryable(e, "commitTxn(" + rqst + ")");` remained
that looks very confusing when we have mix of strategies